### DISCUSSION TITLE
How do i set a permanent password in my strategy?

### DISCUSSION THREAD

**Discussion by CLU91** [rustdesk/rustdesk-server-pro#222] [Q&A] [answered]
Hello,

i created a strategy for all my clients to push defined settings. How do i set a permanent password in the web client? If i choose the option "permanent password" i am not able to connect to the machines anymore. I guess because there is no password set. 

I would not like to set the password on each machine by hand....

regards

**Comment by dinger1986** [comment]
How are you pushing the clients out?

There's deployment script examples on the docs https://rustdesk.com/docs/en/self-host/client-deployment/

These passwords can be set to something if you would rather

**Comment by CLU91** [comment]
I installed by hand and thought you can set it in Webclient for it to be pushed to the machine. But i will definately not put a secure password in a powershell script that will be pushed throughout the network. Thats ridiclious..... if there is no password encryption i will do it by hand on each machine.

**Comment by rustdesk** [comment, marked as answer]
I totally undersand this requirement and the convinience to you.
But I do not think this is a good practise, also I do not see this on any other vendors. 
Techinically, it is eays to add, but we can not take this risk.

However, I am still open to this, but need to listen to more voice from the other users before taking action.

-----

Our custom client generator suppor preset password, and there is a warning.

![image](https://github.com/rustdesk/rustdesk-server-pro/assets/71636191/d84c872f-fb5d-4537-9353-de6cc3d7b5ca)

**Comment by Sebshino** [comment]
hello, with TV quicksupport we can define a generic password (this is when generating the quicksupport)

**Comment by User35123** [comment]
Same for "AnyDesk" (anydesk.com). There you can also define a generic password in custom clients setup.

I having the same problem. I need to setup a default password for all my clients without writing the password as plain text into my script where everybody can see it. This is currently a real problem...

**Comment by CakeOD36** [comment]
I wanted to follow up on this request.   Client management would be more secure and flexible where the device password is assigned by the strategy.   This would minimize the number of deployment builds required and keep the password out of a plain-text file as others note.   This also allows for rotation of the device password when required.

**Comment by rustdesk** [comment]
https://www.reddit.com/r/rustdesk/comments/1bqg7p8/device_password_management/
